pkg/engine/task/task_toggle.go

```diff
@@ -0,0 +1,65 @@
+package task
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+const (
+	toggleTaskError = "ToggleTaskError"
+)
+
+// ToggleTask will apply or delete a set of given resources to the cluster based on value of Parameter. See Run method for more details.
+type ToggleTask struct {
+	Name      string
+	Parameter string
+	Resources []string
+}
+
+func (tt ToggleTask) Run(ctx Context) (bool, error) {
+	// 1. - Get the task to run
+	task, err := tt.intermediateTask(ctx)
```

Comment:
I know this was already changed from `getTask` but `intermediateTask` it is not and it is confusing.  intermediate is defined as "coming between two things "... this isn't IMO

This is a task selection...  a determination of which task to run.
Things that seem better `delegateTask` or `selectTask` or just `task`... perhaps `taskImpl`

Reply:
`delegateTask` sounds good